Allow registering the deviceApiKey in a non-compatible OS by making the apiKey equal the deviceApiKey, and add an fsync to all config.json writes

pcarranzav opened the pull request:
Change-Type: patch
Signed-off-by: Pablo Carranza Velez <pablo@resin.io>

Page- commented on src/bootstrap.coffee:
```diff
@@ -64,11 +73,15 @@ exchangeKey = ->
-			request.postAsync("#{config.apiEndpoint}/api-key/device/#{device.id}/device-key?apikey=#{userConfig.apiKey}")
+			# We found the device, we can try to register a working device key for it
+			userConfig.deviceApiKey ?= deviceRegister.generateUniqueKey()
+			request.postAsync("#{config.apiEndpoint}/api-key/device/#{device.id}/device-key?apikey=#{userConfig.apiKey}", { json: true, body: apiKey: userConfig.deviceApiKey })
```
Can you split the options object onto new lines please, it's really easy to miss that we pass extra options atm as it's after a long string

pcarranzav replied:
Done

pcarranzav commented:
@Page- I made a few changes:
* Fixed getting the osVersion before checking if the OS has deviceApiKey support during provisioning
* Removed the `userConfig.deviceApiKey = body` when doing the key exchange, as this PR will depend on the resin-api PR anyways.

pcarranzav commented:
Connects to https://github.com/resin-io/hq/pull/938/files
